components/function-controller/pkg/apis/serverless/v1alpha2/function_defaults.go

```diff
@@ -68,19 +68,29 @@ func (spec *FunctionSpec) defaultReplicas(config *DefaultingConfig, fn *Function
 	defaultingConfig := config.Function.Replicas
 	replicasPreset := mergeReplicasPreset(fn, defaultingConfig.Presets, defaultingConfig.DefaultPreset)
 
+	if spec.Replicas == nil && spec.ScaleConfig == nil && replicasPreset.Max == replicasPreset.Min {
```

Comment:
I would change it a little bit:
if ScaleConfig != nil -> defaultScaleConfig()
else
check and default if needed the replicas field.

We will end up in situation like this:
if nothing is set, we set the simplest field `replicas`